pkg/daemon/update.go

```diff
@@ -303,16 +299,10 @@ func (dn *Daemon) updateFiles(oldConfig, newConfig *mcfgv1.MachineConfig) error
 	if err := dn.writeFiles(newConfig.Spec.Config.Storage.Files); err != nil {
 		return err
 	}
-
 	if err := dn.writeUnits(newConfig.Spec.Config.Systemd.Units); err != nil {
 		return err
 	}
-
-	if err := dn.deleteStaleData(oldConfig, newConfig); err != nil {
```

Comment:
nit: if in future you add another step after deleteStaleData, you'll have to bring back this if... The current setup keeps it clearer imo.

Reply:
Boils down to preference for me, if you prefer it that way I'll change it back